src/utils/isPlainObject.js

```diff
@@ -1,12 +1,23 @@
+const fnToString = (fn) => Function.prototype.toString.call(fn);
+
 /**
  * @param {any} obj The object to inspect.
  * @returns {boolean} True if the argument appears to be a plain object.
  */
 export default function isPlainObject(obj) {
-  if (!obj) {
+  if (!obj || typeof obj !== 'object') {
     return false;
   }
 
-  return typeof obj === 'object' &&
-         Object.getPrototypeOf(obj) === Object.prototype;
+  const proto = typeof obj.constructor === 'function' ? Object.getPrototypeOf(obj) : Object.prototype;
```

Comment:
Please change `const` to `var`, we're maintaining Flow-friendliness for now. (I'll alter ESLint rule later.)